### ISSUE TITLE
Running the `quarto preview` command in a docusaurus project opens the wrong port

### ISSUE THREAD

**Issue by eitsupi** [quarto-dev/quarto-cli#2684]
### Bug description

I tried the latest quarto cli and it opened the wrong port (6582).
The correct port seems to be the one shown on the command line (3512).
(Note that the command was executed on the terminal in the VSCode Dev Containers.)

![image](https://user-images.githubusercontent.com/50911393/193285964-393554e0-4173-4f92-b520-28f6e7b4a47f.png)

```sh
$ quarto check

[✓] Checking Quarto installation......OK
      Version: 1.2.171
      Path: /opt/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.6
      Path: /usr/local/bin/python3
      Jupyter: 4.11.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by eitsupi** [comment]
@jjallaire Sorry to bother you, but is this the intended behavior?

**Comment by eitsupi** [comment]
Thanks for your response.
I just ran `npx create-docusaurus@latest . classic` to create a sample project of docusaurus and then ran `quarto preview` on vscode integrated terminal.

**Comment by jjallaire** [comment]
It all works as expected for me with:

```bash
$ npx create-docusaurus@latest . classic
$ quarto preview
```

**Comment by jjallaire** [comment]
I don't know how to control that popup (it seems to be auto-detected by VS Code, that's not something the extension is doing). If someone here knows a way to designate that a port shouldn't result in a popup please let me know!

**Comment by eitsupi** [comment]
Perhaps vscode automatically detects the port being used?
Since exiting preview with Ctrl+C also ends the two port transfer, does it appear that the two ports are actually in use during preview?

![image](https://user-images.githubusercontent.com/50911393/194539203-7489ff6c-b8e0-4536-bcc3-a46c47d2f0f7.png)
![image](https://user-images.githubusercontent.com/50911393/194539338-78051a5e-979a-4379-a1dd-8ffca674bdd6.png)

**Comment by jjallaire** [comment]
Yes, there are indeed two ports in use during preview. One is Docusaurus and the other is a "control channel" for VS Code or RStudio to request renders (or server termination). That said, since you are running `quarto preview` directly from the terminal this control channel isn't doing anything (so I can probably clean that up pretty easily, should have that in soon).

I am curious though for scenarios where VS Code initiates the render, what is user-level problem/confusion created -- is it just showing two ports in use or is there something else that can trip people up?

**Comment by eitsupi** [comment]
Thanks for the explanation.
I don't see any problem other than I see a non-functioning port.

**Comment by eitsupi** [comment]
It works fine now. Thanks!
